src/main/resources/reference.conf

```diff
@@ -1345,12 +1351,15 @@ resourceTypes = {
       custodian = {
         roleActions = ["manage_schema", "create_datasnapshot", "ingest_data", "soft_delete", "hard_delete", "unlink_snapshot", "list_snapshots", "lock_resource", "unlock_resource", "share_policy::custodian", "share_policy::ingester", "share_policy::snapshot_creator"]
         includedRoles = ["snapshot_creator"]
+        descendantRoles = {
+          datasnapshot = ["steward"]
+        }
       }
       ingester = {
         roleActions = ["read_dataset", "read_data", "ingest_data"]
       }
       snapshot_creator = {
-        roleActions = ["read_dataset", "read_data", "read_policies", "link_snapshot"]
+        roleActions = ["read_dataset", "read_data", "read_policies", "link_snapshot", "add_child", "list_children"]
```

Comment:
I think this should be at the dataset custodian role instead of snapshot_creator. I think we only want to let dataset custodian and stewards update this inheritSteward role and set children.

Reply:
Is there an add_child endpoint in Sam? I only see setParent. Do we need to instead add the "set_parent" role to datasnapshots?

Reply:
Thanks for talking through this with me:
- "add_child" needs to be on the snapshot_creator role so that on snapshot create, the new snapshot can be added as a child to the dataset. We expect for snapshot_creators to perform this option. 
- "add_child" is checked on "create_with_parent" or "set_parent", so we need both. 

However, I'm not sure about leaving "list_children" on the snapshot_creator. This will effectively grant the snapshot_creator with "list_snapshots" permissions, which was previously reserved for dataset stewards and custodians. Can we move "list_children" up to custodian level instead?

Reply:
> However, I'm not sure about leaving "list_children" on the snapshot_creator. This will effectively grant the snapshot_creator with "list_snapshots" permissions, which was previously reserved for dataset stewards and custodians. Can we move "list_children" up to custodian level instead?

Yes, I think you're right, a snapshot creator shouldn't need this permission and it could let them have more permissions than they had before.  I'll move it to `custodian`.